feat(linux): Try to start ibus if not running

ermshiperete opened the pull request:
When we try to restart ibus and that fails and ibus isn't running afterwards we directly try to start ibus-daemon.

# User Testing

<details>
<summary>Expand to see user tests</summary>

## Preparations

- install the current alpha version of Keyman
- install a second system keyboard, e.g. French
- install a Keyman keyboard
- verify that the keyboard shows up in the keyboards dropdown

## Tests

### SUITE_INSTALL: Package installation 

- **GROUP_BIONIC**: Ubuntu 18.04 Bionic with Gnome shell and X11
- **GROUP_IMPISH**: Ubuntu 21.10 Impish with Gnome shell and X11

  - when login in, click the cog and select "Ubuntu on Xorg"

![Screenshot from 2022-01-19 18-28-11](https://user-images.githubusercontent.com/181336/150183345-f07c7191-b12b-4739-b4fc-c585148db56c.png)

  - once logged in, open a terminal window and verify that running `echo $XDG_SESSION_TYPE`  outputs `x11`.

- **GROUP_IMPISH_WAYLAND**: Ubuntu 21.10 Impish with Gnome shell and Wayland

  - when login in, click the cog and select "Ubuntu"
  - once logged in, open a terminal window and verify that running `echo $XDG_SESSION_TYPE`  outputs `wayland`.

- **GROUP_WASTA**: Wasta 20.04 with Cinnamon

**TEST_INST_RUN**: Install Keyman packages while ibus is running

- download the Keyman packages for this PR
- install the packages with: 
    ```
    sudo dpkg -i \
    {ibus-keyman,libkmnkbp0}*$(lsb_release -s -c)*$(dpkg --print-architecture).deb \
    {keyman,python3-keyman-config}*$(lsb_release -s -c)*all.deb
    ```
- **verify** that ibus is still running by checking that keyboard still shows up in the keyboards dropdown

**TEST_INST_NOIBUS**: Install Keyman packages while ibus is not running

(skip this test on **Bionic** - `ibus-daemon` gets automatically restarted, so it's not possible to run this test)

- exit ibus by running: `killall ibus-daemon`
- verify that ibus is no longer running: Keyman keyboard should no longer show up in keyboards dropdown. It's OK if the keyboards dropdown icon no longer shows.
- re-install the packages for this PR by running:
    ```
    sudo dpkg -i \
    {ibus-keyman,libkmnkbp0}*$(lsb_release -s -c)*$(dpkg --print-architecture).deb \
    {keyman,python3-keyman-config}*$(lsb_release -s -c)*all.deb
    ```
- **verify** that the installation restarted ibus by checking that the keyman keyboard again shows up in the keyboard dropdown. Note that it is OK if a warning shows up in the terminal window that it can't connect to ibus.

### SUITE_ADD_KEYBOARD: Adding keyboards

- **GROUP_BIONIC**: Ubuntu 18.04 Bionic with Gnome shell and X11
- **GROUP_IMPISH**: Ubuntu 21.10 Impish with Gnome shell and X11
- **GROUP_IMPISH_WAYLAND**: Ubuntu 21.10 Impish with Gnome shell and Wayland
- **GROUP_WASTA**: Wasta 20.04 with Cinnamon

**TEST_ADD_RUN**: Add a new Keyman keyboard while ibus is running

- start Keyman Configuration
- add a new Keyman keyboard
- **verify** that ibus is still running by checking that the new keyboard shows up in the keyboard dropdown

**TEST_ADD_NOIBUS**: Add a new Keyman keyboard while ibus is not running

(skip this test on **Bionic** - `ibus-daemon` gets automatically restarted, so it's not possible to run this test)

- exit ibus by running: `killall ibus-daemon`
- verify that ibus is no longer running: Keyman keyboard should no longer show up in keyboards dropdown.  It's OK if the keyboards dropdown icon no longer shows.
- start Keyman Configuration
- add a new Keyman keyboard
- **verify** that ibus got started by checking that the new keyboard shows up in the keyboard dropdown. Note that it is OK if a warning shows up in the terminal window that it can't connect to ibus.
</details>

MakaraSok commented:
## SUITE_INSTALL: Package installation

* **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell
  - **TEST_INST_RUN: FAILED** The ibus is not running after the package was installed on top of the latest alpha installed in the prep stage.

  Note that there are warning and errors after running the command provided:
```
  keyman@keyman-VirtualBox:~/Downloads/6109/results$ sudo dpkg -i \
> {ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*$(lsb_release -s -c)*.deb
[sudo] password for keyman: 
dpkg: warning: downgrading ibus-keyman from 15.0.184-1~sil1~bionic to 15.0.182-1~PR-6109+bionic11
(Reading database ... 165123 files and directories currently installed.)
Preparing to unpack ibus-keyman_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking ibus-keyman (15.0.182-1~PR-6109+bionic11) over (15.0.184-1~sil1~bionic) ...
Preparing to unpack ibus-keyman_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking ibus-keyman:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package ibus-kmfl.
Preparing to unpack ibus-kmfl_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking ibus-kmfl (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack ibus-kmfl_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking ibus-kmfl:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
dpkg: warning: downgrading keyman from 15.0.184-1~sil1~bionic to 15.0.182-1~PR-6109+bionic11
Preparing to unpack keyman_15.0.182-1~PR-6109+bionic11_all.deb ...
Unpacking keyman (15.0.182-1~PR-6109+bionic11) over (15.0.184-1~sil1~bionic) ...
Selecting previously unselected package kmflcomp.
Preparing to unpack kmflcomp_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking kmflcomp (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack kmflcomp_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking kmflcomp:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmfl0:amd64.
Preparing to unpack libkmfl0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmfl0:amd64 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmfl0:i386.
Preparing to unpack libkmfl0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmfl0:i386 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmflcomp0:amd64.
Preparing to unpack libkmflcomp0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmflcomp0:amd64 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmflcomp0:i386.
Preparing to unpack libkmflcomp0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmflcomp0:i386 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmflcomp-dev:amd64.
Preparing to unpack libkmflcomp-dev_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmflcomp-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmflcomp-dev:i386.
Preparing to unpack libkmflcomp-dev_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmflcomp-dev:i386 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmfl-dev:amd64.
Preparing to unpack libkmfl-dev_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmfl-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package libkmfl-dev:i386.
Preparing to unpack libkmfl-dev_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmfl-dev:i386 (15.0.182-1~PR-6109+bionic11) ...
dpkg: warning: downgrading libkmnkbp0-0:amd64 from 15.0.184-1~sil1~bionic to 15.0.182-1~PR-6109+bionic11
Preparing to unpack libkmnkbp0-0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmnkbp0-0:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.184-1~sil1~bionic) ...
Selecting previously unselected package libkmnkbp0-0:i386.
Preparing to unpack libkmnkbp0-0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmnkbp0-0:i386 (15.0.182-1~PR-6109+bionic11) ...
dpkg: warning: downgrading python3-keyman-config from 15.0.184-1~sil1~bionic to 15.0.182-1~PR-6109+bionic11
Preparing to unpack python3-keyman-config_15.0.182-1~PR-6109+bionic11_all.deb ...
Unpacking python3-keyman-config (15.0.182-1~PR-6109+bionic11) over (15.0.184-1~sil1~bionic) ...
More than one copy of package ibus-keyman:i386 has been unpacked
 in this run !  Only configuring it once.
More than one copy of package ibus-kmfl:i386 has been unpacked
 in this run !  Only configuring it once.
More than one copy of package kmflcomp:i386 has been unpacked
 in this run !  Only configuring it once.
dpkg: dependency problems prevent configuration of ibus-keyman:i386:
 ibus-keyman:i386 depends on sudo.
 ibus-keyman:i386 depends on libc6 (>= 2.4).
 ibus-keyman:i386 depends on libglib2.0-0 (>= 2.37.3).
 ibus-keyman:i386 depends on libgtk-3-0 (>= 3.0.0).
 ibus-keyman:i386 depends on libibus-1.0-5 (>= 1.5.14).
 ibus-keyman:i386 depends on libjson-glib-1.0-0 (>= 0.12.0).
 ibus-keyman:i386 depends on libx11-6.

dpkg: error processing package ibus-keyman:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of ibus-kmfl:i386:
 ibus-kmfl:i386 depends on libc6 (>= 2.4).
 ibus-kmfl:i386 depends on libglib2.0-0 (>= 2.16.0).
 ibus-kmfl:i386 depends on libibus-1.0-5 (>= 1.5.1).
 ibus-kmfl:i386 depends on libx11-6.

dpkg: error processing package ibus-kmfl:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of kmflcomp:i386:
 kmflcomp:i386 depends on libc6 (>= 2.3.4).

dpkg: error processing package kmflcomp:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libkmfl0:i386:
 libkmfl0:i386 depends on libc6 (>= 2.4).

dpkg: error processing package libkmfl0:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmflcomp0:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmflcomp0:i386:
 libkmflcomp0:i386 depends on libc6 (>= 2.11).
 libkmflcomp0:i386 depends on libx11-6.

dpkg: error processing package libkmflcomp0:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmflcomp-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmflcomp-dev:i386:
 libkmflcomp-dev:i386 depends on libkmflcomp0 (= 15.0.182-1~PR-6109+bionic11); however:
  Package libkmflcomp0:i386 is not configured yet.

dpkg: error processing package libkmflcomp-dev:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libkmfl-dev:i386:
 libkmfl-dev:i386 depends on libkmfl0 (= 15.0.182-1~PR-6109+bionic11); however:
  Package libkmfl0:i386 is not configured yet.

dpkg: error processing package libkmfl-dev:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmnkbp0-0:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmnkbp0-0:i386:
 libkmnkbp0-0:i386 depends on libc6 (>= 2.4).
 libkmnkbp0-0:i386 depends on libgcc1 (>= 1:3.0).
 libkmnkbp0-0:i386 depends on libstdc++6 (>= 5.2).

dpkg: error processing package libkmnkbp0-0:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of python3-keyman-config:
 python3-keyman-config depends on keyman-engine; however:
  Package keyman-engine is not installed.

dpkg: error processing package python3-keyman-config (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of keyman:
 keyman depends on python3-keyman-config; however:
  Package python3-keyman-config is not configured yet.

dpkg: error processing package keyman (--install):
 dependency problems - leaving unconfigured
Setting up libkmfl0:amd64 (15.0.182-1~PR-6109+bionic11) ...
Setting up libkmfl-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for desktop-file-utils (0.23-1ubuntu3.18.04.2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for libglib2.0-0:amd64 (2.56.4-0ubuntu0.18.04.9) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (1.9-2) ...
Processing triggers for libc-bin (2.27-3ubuntu1.4) ...
Errors were encountered while processing:
 ibus-keyman:i386
 ibus-kmfl:i386
 kmflcomp:i386
 libkmfl0:i386
 libkmflcomp0:i386
 libkmflcomp-dev:i386
 libkmfl-dev:i386
 libkmnkbp0-0:i386
 python3-keyman-config
 keyman
  ```

  Since the ibus doesn't run at all, another attempt has been made to install the build package of this pr, but then Ubuntu doesn't like it. <br/>
  <img width="300" alt="error after running the three commands to install the build from this PR" src="https://user-images.githubusercontent.com/28331388/150895382-19e51c63-df02-480d-be8e-8e947066143d.png">
  
  - **TEST_INST_NOIBUS: FAILED** The installation doesn't retart the ibus. No Keyman keyboard is seen in the dropdown menu. Here is what's shown up in the terminal. <br/>
  ```
  keyman@keyman-VirtualBox:~/Downloads/6109/results$ sudo dpkg -i \
> {ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*$(lsb_release -s -c)*.deb
(Reading database ... 165039 files and directories currently installed.)
Preparing to unpack ibus-keyman_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking ibus-keyman (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack ibus-keyman_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking ibus-keyman:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack ibus-kmfl_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking ibus-kmfl (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack ibus-kmfl_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking ibus-kmfl:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package keyman.
Preparing to unpack keyman_15.0.182-1~PR-6109+bionic11_all.deb ...
Unpacking keyman (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack kmflcomp_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking kmflcomp (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack kmflcomp_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking kmflcomp:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmfl0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmfl0:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmfl0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmfl0:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmflcomp0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmflcomp0:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmflcomp0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmflcomp0:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmflcomp-dev_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmflcomp-dev:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmflcomp-dev_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmflcomp-dev:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmfl-dev_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmfl-dev:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmfl-dev_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmfl-dev:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmnkbp0-0_15.0.182-1~PR-6109+bionic11_amd64.deb ...
Unpacking libkmnkbp0-0:amd64 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Preparing to unpack libkmnkbp0-0_15.0.182-1~PR-6109+bionic11_i386.deb ...
Unpacking libkmnkbp0-0:i386 (15.0.182-1~PR-6109+bionic11) over (15.0.182-1~PR-6109+bionic11) ...
Selecting previously unselected package python3-keyman-config.
Preparing to unpack python3-keyman-config_15.0.182-1~PR-6109+bionic11_all.deb ...
Unpacking python3-keyman-config (15.0.182-1~PR-6109+bionic11) ...
More than one copy of package ibus-keyman:i386 has been unpacked
 in this run !  Only configuring it once.
More than one copy of package ibus-kmfl:i386 has been unpacked
 in this run !  Only configuring it once.
More than one copy of package kmflcomp:i386 has been unpacked
 in this run !  Only configuring it once.
dpkg: dependency problems prevent configuration of ibus-keyman:i386:
 ibus-keyman:i386 depends on sudo.
 ibus-keyman:i386 depends on libc6 (>= 2.4).
 ibus-keyman:i386 depends on libglib2.0-0 (>= 2.37.3).
 ibus-keyman:i386 depends on libgtk-3-0 (>= 3.0.0).
 ibus-keyman:i386 depends on libibus-1.0-5 (>= 1.5.14).
 ibus-keyman:i386 depends on libjson-glib-1.0-0 (>= 0.12.0).
 ibus-keyman:i386 depends on libx11-6.

dpkg: error processing package ibus-keyman:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of ibus-kmfl:i386:
 ibus-kmfl:i386 depends on libc6 (>= 2.4).
 ibus-kmfl:i386 depends on libglib2.0-0 (>= 2.16.0).
 ibus-kmfl:i386 depends on libibus-1.0-5 (>= 1.5.1).
 ibus-kmfl:i386 depends on libx11-6.

dpkg: error processing package ibus-kmfl:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of kmflcomp:i386:
 kmflcomp:i386 depends on libc6 (>= 2.3.4).

dpkg: error processing package kmflcomp:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libkmfl0:i386:
 libkmfl0:i386 depends on libc6 (>= 2.4).

dpkg: error processing package libkmfl0:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmflcomp0:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmflcomp0:i386:
 libkmflcomp0:i386 depends on libc6 (>= 2.11).
 libkmflcomp0:i386 depends on libx11-6.

dpkg: error processing package libkmflcomp0:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmflcomp-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmflcomp-dev:i386:
 libkmflcomp-dev:i386 depends on libkmflcomp0 (= 15.0.182-1~PR-6109+bionic11); however:
  Package libkmflcomp0:i386 is not configured yet.

dpkg: error processing package libkmflcomp-dev:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of libkmfl-dev:i386:
 libkmfl-dev:i386 depends on libkmfl0 (= 15.0.182-1~PR-6109+bionic11); however:
  Package libkmfl0:i386 is not configured yet.

dpkg: error processing package libkmfl-dev:i386 (--install):
 dependency problems - leaving unconfigured
Setting up libkmnkbp0-0:amd64 (15.0.182-1~PR-6109+bionic11) ...
dpkg: dependency problems prevent configuration of libkmnkbp0-0:i386:
 libkmnkbp0-0:i386 depends on libc6 (>= 2.4).
 libkmnkbp0-0:i386 depends on libgcc1 (>= 1:3.0).
 libkmnkbp0-0:i386 depends on libstdc++6 (>= 5.2).

dpkg: error processing package libkmnkbp0-0:i386 (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of python3-keyman-config:
 python3-keyman-config depends on keyman-engine; however:
  Package keyman-engine is not installed.

dpkg: error processing package python3-keyman-config (--install):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of keyman:
 keyman depends on python3-keyman-config; however:
  Package python3-keyman-config is not configured yet.

dpkg: error processing package keyman (--install):
 dependency problems - leaving unconfigured
Setting up libkmfl0:amd64 (15.0.182-1~PR-6109+bionic11) ...
Setting up libkmfl-dev:amd64 (15.0.182-1~PR-6109+bionic11) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Processing triggers for gnome-menus (3.13.3-11ubuntu1.1) ...
Processing triggers for desktop-file-utils (0.23-1ubuntu3.18.04.2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for libglib2.0-0:amd64 (2.56.4-0ubuntu0.18.04.9) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (1.9-2) ...
Processing triggers for libc-bin (2.27-3ubuntu1.4) ...
Errors were encountered while processing:
 ibus-keyman:i386
 ibus-kmfl:i386
 kmflcomp:i386
 libkmfl0:i386
 libkmflcomp0:i386
 libkmflcomp-dev:i386
 libkmfl-dev:i386
 libkmnkbp0-0:i386
 python3-keyman-config
 keyman
```

## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell**
  - **TEST_ADD_RUN: FAILED** ibus is still not running. Keyman app is visible as a result of the previous reinstallation of the build package, but added keyboards don't shown up in the dropdown. In fact, the dropdown in nowhere to be seen.
  - **TEST_ADD_NOIBUS: FAILED** ibus stays cool and nothing shows up as indicated.

MakaraSok commented:
## SUITE_INSTALL: Package installation

* **GROUP_WASTA: Wasta 20.04**
In the prep, the latest alpha was install using the method provided at https://keyman.com/downloads/pre-release/. The second system keyboard was added and it shows in the menu, but separate from the one with Keyman keyboard. <br/>
A Keyman keyboard was installed, howver it doesn't show up in the menu. See the screenshot below. <br/>
<img width="300" src="https://user-images.githubusercontent.com/28331388/150903352-8200b6b3-56d7-4051-af4e-49bcb7582c40.png"> <br/>
  - **TEST_INST_RUN: PASSED** ibus seems to be running after the installation. <br/>
  <img width="300" src="https://user-images.githubusercontent.com/28331388/150904220-842182a5-b24e-41f0-b526-78c389cb6746.png"> <br/>
  - **TEST_INST_NOIBUS: PASSED?** after running the command, the menu disappears, but the already opened Keyman Configuration window stays. Reinstalling ibus does bring it back to life, but again the keyboards installed doesn't reflect in the menu. <br/>
  <img width="300" src="https://user-images.githubusercontent.com/28331388/150905767-61ecc1cc-f684-420a-bccc-e0b3913b26b9.png"> <br/>
## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_WASTA: Wasta 20.04**
  - **TEST_ADD_RUN: FAILED** The keyboard doesn't show up automatically in the menu, but one has to go to the ibus preference to add the keyboard from the list of input method. <br/>
  <img width="300" src="https://user-images.githubusercontent.com/28331388/150904912-64c0c6bf-a476-45b5-8155-8ba70e4ae855.png"> <br/>
  - **TEST_ADD_NOIBUS: PASSED?** run `km-config` and install a new keyboard does bring ibus back to life, but the recently installed keyboard doesn't show in the menu. <br/>
  <img width="300" src="https://user-images.githubusercontent.com/28331388/150905329-5868bb06-f2da-4d6b-87f4-a067e8de7eeb.png"> <br/>

MakaraSok commented:
## SUITE_INSTALL: Package installation
In the preparation stage, when trying to install the latest alpha version of Keyman,here is what's returned:
```
sudo apt install keyman onboard-keyman
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package onboard-keyman is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
```
* **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**
  - **TEST_INST_RUN: BLOCKED**
  - **TEST_INST_NOIBUS: BLOCKED**

## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**
  - **TEST_ADD_RUN: BLOCKED**
  - **TEST_ADD_NOIBUS: BLOCKED**

ermshiperete commented:
Bionic: sorry, my instructions were incomplete. I updated them. Previously it tried to install both 64- and 32-bit versions at the same time.

The installation command has to be:

```
sudo dpkg -i \
{ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*\
$(lsb_release -s -c)*$(dpkg --print-architecture).deb
```

@keymanapp-test-bot retest GROUP_BIONIC

ermshiperete commented:
Impish: make sure you follow the instructions in https://help.keyman.com/knowledge-base/103 before trying to install Keyman, in particular run the two commands:

```
wget -qO - http://linux.lsdev.sil.org/downloads/sil-testing.gpg | sudo apt-key add -
sudo add-apt-repository "deb http://linux.lsdev.sil.org/ubuntu $(lsb_release -sc)-experimental main"
```

@MakaraSok Can you retry please?

@keymanapp-test-bot retest GROUP_IMPISH

ermshiperete commented:
@keymanapp-test-bot retest SUITE_ADD_KEYBOARD GROUP_BIONIC all

ermshiperete commented:
@keymanapp-test-bot retest SUITE_ADD_KEYBOARD GROUP_IMPISH all

ermshiperete commented:
Wasta-Linux: seems that's a bug that sneaked in independent of this PR. See https://github.com/keymanapp/keyman/issues/6158.

MakaraSok commented:
## SUITE_INSTALL: Package installation
Prep stage: everything set as indicated and up to the expectation. No error here.
  * **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell**
    - **TEST_INST_RUN: FAILED** The package cannot be found in the newly built package. <br/>
```
keyman@keyman-VirtualBox:~/Downloads/6109/results$ sudo dpkg -i \
> {ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*\
> $(lsb_release -s -c)*$(dpkg --print-architecture).deb
[sudo] password for keyman: 
dpkg: error: cannot access archive 'keyman*bionic*amd64.deb': No such file or directory
```
<br/>
    - **TEST_INST_NOIBUS: BLOCKED**

  * **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**
  After installing the latest alpha and a Keyman keyboard, the keyboard doesn't show. Only the system ones are visible in the drop down menu.

  The return when running sanity check: <br/>
```
keyman@keyman-VirtualBox:~$ ps -C ibus-daemon -f
UID          PID    PPID  C STIME TTY          TIME CMD
```
<br/>
  The Keyman keyboard appears after restarting. 
<br/>

    * **TEST_INST_RUN: FAILED** The same error experienced in Bionic. <br/>
```
keyman@keyman-VirtualBox:~/Downloads/6109/archive-3/results$ sudo dpkg -i \
{ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*\
$(lsb_release -s -c)*$(dpkg --print-architecture).deb
[sudo] password for keyman: 
dpkg: error: cannot access archive 'keyman*impish*amd64.deb': No such file or directory
keyman@keyman-VirtualBox:~/Downloads/6109/archive-3/results$ sudo dpkg -i \
{ibus-keyman,ibus-kmfl,keyman,kmflcomp,libkmfl,libkmnkbp0,python3-keyman-config}*\
$(lsb_release -s -c)*$(dpkg --print-architecture).deb
dpkg: error: cannot access archive 'keyman*impish*amd64.deb': No such file or directory
keyman@keyman-VirtualBox:~/Downloads/6109/archive-3/results$ 
```
<br/>
    * **TEST_INST_NOIBUS: BLOCKED**

## SUITE_ADD_KEYBOARD: Adding keyboards

  * **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell**

    * **TEST_ADD_RUN: BLOCKED**
    * **TEST_ADD_NOIBUS: BLOCKED**
    
  * **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**

    * **TEST_ADD_RUN: BLOCKED**
    * **TEST_ADD_NOIBUS: BLOCKED**

ermshiperete commented:
@keymanapp-test-bot retest SUITE_ADD_KEYBOARD GROUP_BIONIC all

ermshiperete commented:
@keymanapp-test-bot retest SUITE_ADD_KEYBOARD GROUP_IMPISH all

ermshiperete commented:
@keymanapp-test-bot retest SUITE_INSTALL GROUP_IMPISH all

ermshiperete commented:
@keymanapp-test-bot retest SUITE_INSTALL GROUP_BIONIC all

ermshiperete commented:
@keymanapp-test-bot ignore SUITE_ADD_KEYBOARD GROUP_WASTA TEST_ADD_RUN

MakaraSok commented:
## SUITE_INSTALL: Package installation
Prep: everything is OK. The latest alpha was installed, the keyboards added (both system and Keyman) are visible iin the dropdown menu.
<img width="150" src="https://user-images.githubusercontent.com/28331388/151284566-a4efd41b-3f53-4a30-8cf1-1ed4e2d97d28.png">

  * **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell**
    - **TEST_INST_RUN: PASSED** ibus is indeed still running
    - **TEST_INST_NOIBUS: FAILED** ibus is still running after running this command `killall ibus-daemon` <br/>
    <img width="150" src="https://user-images.githubusercontent.com/28331388/151285032-475fa62b-7ca8-472e-ba95-59a130f63266.png">

## SUITE_ADD_KEYBOARD: Adding keyboards

  * **GROUP_BIONIC: Ubuntu 18.04 Bionic with Gnome shell**
    * **TEST_ADD_RUN: PASSED** 
    * **TEST_ADD_NOIBUS: FAILED** ibus is still running aftering being killed.
    <img width="150" src="https://user-images.githubusercontent.com/28331388/151287723-deb5320d-cb45-414a-b835-d2cbad4a80fa.png">

    This error popped up from time to time during testing.
    <img width="150" src="https://user-images.githubusercontent.com/28331388/151287989-c5f91eef-74be-46cd-bc83-f193cc8c9077.png">

    It looks like ibus crashes:
```
    keyman@keyman-VirtualBox:/var/crash$ dir
_usr_bin_gnome-shell.121.crash	_usr_bin_ibus-daemon.1000.crash
```

MakaraSok commented:
## SUITE_INSTALL: Package installation
Prep: The latest alpha was installed, the Keyman keyboard added are not visible in the dropdown menu.
<img width="300" alt="impish" src="https://user-images.githubusercontent.com/28331388/151304965-43ded78b-36f2-4bfe-b990-8d0fd3473af5.png">

```
keyman@keyman-VirtualBox:~$ ps -C ibus-daemon -f
UID          PID    PPID  C STIME TTY          TIME CMD
```

After a restart, the Keyman keyboard does appear.

  * **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**
    * **TEST_INST_RUN: PASSED?** ibus doesn't seem to be running. No Keyman keyboard is visible in the dropdown menu, until restarting Ubuntu.
    * **TEST_INST_NOIBUS: PASSED?** reinstalling the package doesn't activate ibus. No Keyman keyboard is visible in the dropdown menu, until Ubuntu restarts.
    <img width="300" alt="impish" src="https://user-images.githubusercontent.com/28331388/151307932-93e92c5b-85e2-47d4-9059-e25c09dde1f4.png">

## SUITE_ADD_KEYBOARD: Adding keyboards
    
  * **GROUP_IMPISH: Ubuntu 21.10 Impish with Gnome shell and X11**
    * **TEST_ADD_RUN: PASSED** the newly installed Keyman keyboard is visible in the dropdown menu.
    * **TEST_ADD_NOIBUS: PASSED** after killing the ibus, installing a keyboard from within Keyman Configuration would reactivate ibus and all keyboards are shown in the dropdown menu.
    <img width="300" alt="impish" src="https://user-images.githubusercontent.com/28331388/151308733-c0a14ffd-bf9f-4a22-8068-6e5726232a89.png">

ermshiperete commented:
> * **TEST_INST_NOIBUS: FAILED** ibus is still running after running this command `killall ibus-daemon`

@MakaraSok Can you please show me the output of `ps -ef | grep ibus`?

bharanidharanj commented:
## SUITE_INSTALL: Package Installation

  * **GROUP_IMPISH**: Ubuntu 21.10 Impish with Gnome shell and X11
       * **TEST_INST_RUN PASSED**: Verified this issue (with ibus running) and the Keyboard still shows up in the keyboards dropdown list in the task bar.
       * **TEST_INST_NOIBUS PASSED**: Tested this issue (with no ibus running) and the Keyboard still shows up in the Keyboards dropdown(Ubuntu onXorg)

bharanidharanj commented:
* **GROUP_WASTA**: **Wasta 20.04**
     * **TEST_INST_RUN: BLOCKED** 
     *  **TEST_INST_NOIBUS: BLOCKED** 

 <img width="300" src="https://user-images.githubusercontent.com/19683143/152528273-73fa1ab2-b9d4-4e26-b0b3-2554bb69062a.PNG">

bharanidharanj commented:
## SUITE_ADD_KEYBOARD: Adding keyboards

- GROUP_WASTA: Wasta 20.04

  * **TEST_ADD_NOIBUS (FAILED):** Able to see the new keyboard shows up in the keyboard dropdown, however I am getting Warning message that 'could not find connected bus'.

<img width="300" src="https://user-images.githubusercontent.com/19683143/152541509-1aba84be-6fc9-4f56-8df3-12e59c7033a2.PNG">

ermshiperete commented:
@keymanapp-test-bot retest GROUP_WASTA

ermshiperete commented:
@bharanidharanj When I looked into the failing test on Impish, I noticed a problem when running with Wayland, so I fixed that. Unfortunately that has a bigger impact, so you'll have to retest all groups again. I'll also added a new group to test with Wayland. Thanks!

Note that for the NOIBUS tests it is OK if a warning shows up in the terminal window that it can't connect to ibus.

@keymanapp-test-bot retest all

ermshiperete commented:
Updated the test instructions. Because we no longer build the kmfl packages, the command to install Keyman has to be:

```
sudo dpkg -i \
{ibus-keyman,libkmnkbp0}*$(lsb_release -s -c)*$(dpkg --print-architecture).deb \
{keyman,python3-keyman-config}*$(lsb_release -s -c)*all.deb
```

bharanidharanj commented:
## SUITE_INSTALL: Package Installation

  * **GROUP_IMPISH**: **Ubuntu 21.10 Impish with Gnome shell and X11**
       * **TEST_INST_RUN: PASSED** Verified this issue with Keyman 15.0.193-alpha and the Keyboard still shows up in the keyboards dropdown list in the task bar.( Ubuntu onXorg) 
       * **TEST_INST_NOIBUS: PASSED** Tested this issue Keyman 15.0.193-alpha and the Keyboard still shows up in the Keyboards dropdown.(Ubuntu onXorg) 

 ## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_IMPISH**: **Ubuntu 21.10 Impish with Gnome shell and X11**

  * **TEST_ADD_RUN (PASSED):** Verified that the ibus is still running after the new keyboard shows up in the keyboard dropdown.
  * **TEST_ADD_NOIBUS (PASSED):** Able to see the new keyboard shows up in the keyboard dropdown, (after removed the ibus)


## SUITE_INSTALL: Package Installation

* **GROUP_IMPISH**: **GROUP_IMPISH_WAYLAND: Ubuntu 21.10 Impish with Gnome shell and Wayland**

  * **TEST_INST_RUN: PASSED** Verified this issue with 15.0.193-alpha and the Keyboard still shows up in the keyboards dropdown list in the task bar. 
   * **TEST_INST_NOIBUS: FAILED** Tested this issue with 15.0.193-alpha and the Keyboard  does not shows up in the Keyboards dropdown.


## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_IMPISH**: **Ubuntu 21.10 Impish with Gnome shell and Wayland**

  * **TEST_ADD_RUN (PASSED):** Verified that the ibus is still running after the new keyboard shows up in the keyboard dropdown.
  * **TEST_ADD_NOIBUS (PASSED):** Able to see the new keyboard shows up in the keyboard dropdown,

bharanidharanj commented:
> @bharanidharanj It's acceptable if the keyboards icon disappears after killing ibus-daemon. I updated the instructions.
> 
> Thanks for finding the problem with Impish. I fixed it; it should be available in the 15.0.194 PR build.
> 
> @keymanapp-test-bot retest SUITE_INSTALL GROUP_WASTA TEST_INST_NOIBUS, SUITE_ADD_KEYBOARD GROUP_WASTA TEST_ADD_NOIBUS, SUITE_INSTALL GROUP_IMPISH_WAYLAND TEST_INST_NOIBUS

@ermshiperete  Okay, I will retest this in 15.0.194 build.

bharanidharanj commented:
## SUITE_INSTALL: Package Installation

* **GROUP_IMPISH_WAYLAND**: Ubuntu 21.10 Impish with Gnome shell and Wayland

   * **TEST_INST_RUN: PASSED** Verified this issue with 15.0.194-alpha and the Keyboard still shows up in the keyboards dropdown list in the task bar. 
   * **TEST_INST_NOIBUS: PASSED** Tested this issue with 15.0.194-alpha and the Keyboard  still shows up in the Keyboards dropdown.



## SUITE_INSTALL: Package Installation

  * **GROUP_WASTA**: **Wasta 20.04 with Cinnamon**

       * **TEST_INST_RUN (PASSED):  Verified this issue with Keyman 15.0.194-alpha and the Keyboard still shows up in the keyboards dropdown list in the task bar. 
       * **TEST_INST_NOIBUS (PASSED): Tested this issue with Keyman 15.0.194 and it is working as expected. 

## SUITE_ADD_KEYBOARD: Adding keyboards

* **GROUP_WASTA*: **Wasta 20.04 with Cinnamon**

  * **TEST_ADD_RUN (PASSED):** Verified this issue with Keyman 15.0.194-alpha and the ibus is still running after the new keyboard shows up in the keyboard dropdown.
  * **TEST_ADD_NOIBUS (PASSED):** Verified this issue with Keyman 15.0.194-alpha and this is working fine. 


## SUITE_INSTALL: Package Installation

   * **GROUP_BIONIC**: **Ubuntu 18.04 Bionic with Gnome shell and X11**

      * **TEST_INST_RUN (PASSED):  Verified this issue with Keyman 15.0.193-alpha and the Keyboard still shows up in the keyboards dropdown list in the task bar. 
       * **TEST_INST_NOIBUS (SKIPPED):  Skipped this test as per eberhard suggestion. 


## SUITE_ADD_KEYBOARD: Adding keyboards

 * **GROUP_BIONIC**: **Ubuntu 18.04 Bionic with Gnome shell and X11**

   * **TEST_ADD_RUN (PASSED):** Verified this issue with Keyman 15.0.194-alpha and it is working as expected.

keyman-server commented:
Changes in this pull request will be available for download in [Keyman version 15.0.194-alpha](https://keyman.com/downloads/releases/alpha/15.0.194)
